### PR TITLE
IGNITE-20513 Use NativeType as the type of the system view column

### PR DESCRIPTION
**Pull request by korlov42**
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/system-view/src/main/java/org/apache/ignite/internal/systemview/utils/SystemViewUtils.java [hunk 1]
```diff
@@ -49,7 +53,7 @@ public static CreateSystemViewCommand toSystemViewCreateCommand(SystemView<?> vi
                     ColumnParams.builder()
                             .name(((NodeSystemView<?>) view).nodeNameColumnAlias())
                             .type(ColumnType.STRING)
-                            .length(CatalogUtils.DEFAULT_VARLEN_LENGTH)
+                            .length(NODE_NAME_FIELD_LENGTH)
```
Comment (lowka) [inline modules/system-view/src/main/java/org/apache/ignite/internal/systemview/utils/SystemViewUtils.java:56]: Maybe we should set nullability of node alias column to `false`?

File: modules/system-view/src/main/java/org/apache/ignite/internal/systemview/utils/SystemViewUtils.java [hunk 2]
```diff
@@ -66,13 +70,14 @@ public static CreateSystemViewCommand toSystemViewCreateCommand(SystemView<?> vi
     }
 
     private static ColumnParams systemViewColumnToColumnParams(SystemViewColumn<?, ?> column) {
-        NativeTypeSpec typeSpec = NativeTypeSpec.fromClass(column.type());
+        NativeType type = column.type();
+
+        NativeTypeSpec typeSpec = type.spec();
 
         Builder builder = ColumnParams.builder()
                 .name(column.name())
```
Comment (lowka) [inline modules/system-view/src/main/java/org/apache/ignite/internal/systemview/utils/SystemViewUtils.java:78]: And maybe we should set nullability of standard system view columns to `false`?

File: modules/system-view/src/main/java/org/apache/ignite/internal/systemview/utils/SystemViewUtils.java (hunk 1 above)
Comment (korlov42) [reply modules/system-view/src/main/java/org/apache/ignite/internal/systemview/utils/SystemViewUtils.java:56]: good catch! fixed

File: modules/system-view/src/main/java/org/apache/ignite/internal/systemview/utils/SystemViewUtils.java (hunk 2 above)
Comment (korlov42) [reply modules/system-view/src/main/java/org/apache/ignite/internal/systemview/utils/SystemViewUtils.java:79]: good catch! fixed

**Review by lowka — APPROVED** [review]

**Review by zstan — APPROVED** [review]

**Pull request merged**